src/lib/Microsoft.Health.Fhir.Ingest/Data/MeasurementGroupToCompressedEventMessageAsyncCollector.cs

```diff
@@ -0,0 +1,99 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs;
+using EnsureThat;
+using Microsoft.Health.Fhir.Ingest.Service;
+using Microsoft.Health.Logging.Telemetry;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Ingest.Data
+{
+    public class MeasurementGroupToCompressedEventMessageAsyncCollector :
+        IEnumerableAsyncCollector<IMeasurement>
+    {
+        private readonly IEventHubMessageService _eventHubService;
+        private readonly IHashCodeFactory _hashCodeFactory;
+        private readonly ITelemetryLogger _telemetryLogger;
+
+        public MeasurementGroupToCompressedEventMessageAsyncCollector(
+            IEventHubMessageService eventHubService,
+            IHashCodeFactory hashCodeFactory,
+            ITelemetryLogger telemetryLogger)
+        {
+            _eventHubService = EnsureArg.IsNotNull(eventHubService, nameof(eventHubService));
+            _hashCodeFactory = EnsureArg.IsNotNull(hashCodeFactory, nameof(hashCodeFactory));
+            _telemetryLogger = EnsureArg.IsNotNull(telemetryLogger, nameof(telemetryLogger));
+        }
+
+        public async Task AddAsync(IMeasurement item, CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
I believe this is only used in the in the legacy code (Azure Function).  Long term it would be best to split implementations.  Have the simple one for backwards compatibility in the 

I will create a user story to collect this and other clean up refactors for a later PR.